project/routes.py

```diff
@@ -75,6 +76,14 @@ def add(item, user_id):
 
         # add item to database
         add_item(item, quantity, price, user_id)
+        flash("Item added", category="success")
+
+        budget_id = fetch_budget_id(user_id)
+        budget = fetch_budget(budget_id)
+        sub_budget(float(price), budget_id)
+
+        if budget.amount < 0:
```

Comment:
Maybe for a future feature to make it so that it prevents the user from fulfilling their action